src/qibocal/protocols/characterization/resonator_spectroscopy.py

```diff
@@ -60,33 +61,51 @@ class ResonatorSpectroscopyResults(Results):
     """Readout amplitude for each qubit."""
 
 
-class ResonatorSpectroscopyData(DataUnits):
-    """ResonatorSpectroscopy acquisition outputs."""
+ResSpecType = np.dtype(
+    [("freq", np.float64), ("msr", np.float64), ("phase", np.float64)]
+)
+"""Custom dtype for resonator spectroscopy."""
 
-    def __init__(self, resonator_type, power_level=None, amplitude=None):
-        super().__init__(
-            "data",
-            {"frequency": "Hz"},
-            options=["qubit"],
-        )
-        self._resonator_type = resonator_type
-        self._power_level = power_level
-        self._amplitude = amplitude
 
-    @property
-    def resonator_type(self):
-        """Type of resonator"""
-        return self._resonator_type
+@dataclass
+class ResonatorSpectroscopyData:
+    """Data structure for resonator spectroscopy."""
 
-    @property
-    def power_level(self):
-        """Resonator spectroscopy power level"""
-        return self._power_level
+    power_level: PowerLevel
+    """Power regime of the resonator."""
+    resonator_type: str
+    """Resonator type."""
+    amplitudes: list
+    """Amplitudes provided by the user."""
+    data: Dict[QubitId, npt.NDArray[ResSpecType]] = field(default_factory=dict)
+    """Raw data acquired."""
+
+    def load_qubit(self, qubit, freq, msr, phase):
+        """Store output for single qubit."""
+        ar = np.empty(freq.shape, dtype=ResSpecType)
+        ar["freq"] = freq
+        ar["msr"] = msr
+        ar["phase"] = phase
+        self.data[qubit] = np.rec.array(ar)
 
     @property
-    def amplitude(self):
-        """Readout pulse amplitude common for all qubits"""
-        return self._amplitude
+    def qubits(self):
+        """Access qubits from data structure."""
+        return [q for q in self.data]
+
+    def __getitem__(self, qubit):
+        return self.data[qubit]
+
+    def save(self, path):
+        """Store results."""
+        self.power_level = self.power_level.name
+        my_dict = asdict(self)
+        my_dict.pop("data")
+        with open(path / "conf.yml", "w") as outfile:
+            yaml.dump(my_dict, outfile)
```

Comment:
I know I said JSON, but it will be just the same, so I'll write for YAML because it's the current one, and the proposal will be slightly closer:
```suggestion
        (path / "conf.yml").write_text(yaml.dump(my_dict))
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text